commons/c2cgeoportal_commons/alembic/main/678f88c7ad5e_add_vector_tiles_layers_table.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2019, Camptocamp SA
+# All rights reserved.
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+
+# 1. Redistributions of source code must retain the above copyright notice, this
+#    list of conditions and the following disclaimer.
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies,
+# either expressed or implied, of the FreeBSD Project.
+
+"""Add vector tiles layers table
+
+Revision ID: 678f88c7ad5e
+Revises: 028477929d13
+Create Date: 2019-09-27 13:51:08.255601
+"""
+
+from alembic import op
+from c2c.template.config import config
+from sqlalchemy import Column, ForeignKey, Integer, Unicode
+
+
+# revision identifiers, used by Alembic.
+revision = '678f88c7ad5e'
+down_revision = '028477929d13'
+branch_labels = None
+depends_on = None
+schema = config['schema']
+staticschema = config['schema_static']
+
+
+def upgrade():
+    op.create_table(
+        'layer_vectortiles',
+        Column('id', Integer, ForeignKey(schema + '.layer.id'), primary_key=True),
+        Column('style', Unicode, nullable=False),
+        Column('xyz', Unicode),
```

Comment:
xyz is optional, it is used for raster vector tiles (Luxembourg use it for 3D, but other project may not need it). But I will explicitly add Nullable=True for xyz.
style is mandatory.

Reply:
I'm not completely aware of vector tiles, but I suppose we need an url to get the tiles, do I miss something ?

Reply:
Hi @arnaud-morvan, a vector tile layer is much more than an OL layer:
- it needs some style definitions;
- it can reference several data urls;
- but also fonts and sprites;
- it can reference raster layers (! like shading...).

All this is encapsulated in the "style" resource.
Here the style can be either an URL to the style, or the style content.

To support browsers or non-browsers (print, Cesium3D) not working with vector tiles you can optionally specify an [XYZ layer](https://openlayers.org/en/latest/apidoc/module-ol_source_XYZ-XYZ.html) to use instead.

Ping me if you have other questions.

Reply:
Ok, thanks @gberaudo for the explanations.